pkg/agent/openflow/network_policy_test.go

```diff
@@ -282,91 +292,286 @@ func TestInstallPolicyRuleFlows(t *testing.T) {
 }
 
 func TestBatchInstallPolicyRuleFlows(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	c = prepareClient(ctrl)
-	c.nodeConfig = &config.NodeConfig{PodIPv4CIDR: podIPv4CIDR, PodIPv6CIDR: nil}
-	c.ipProtocols = []binding.Protocol{binding.ProtocolIP}
-	defaultAction := crdv1alpha1.RuleActionAllow
-	priorityRule2 := uint16(10000)
-
-	ruleID1 := uint32(10)
-	rule1 := &types.PolicyRule{
-		Direction: v1beta2.DirectionOut,
-		From:      parseAddresses([]string{"192.168.1.40", "192.168.1.50"}),
-		Action:    &defaultAction,
-		To:        parseAddresses([]string{"0.0.0.0/0"}),
-		FlowID:    ruleID1,
-		TableID:   EgressRuleTable,
-		PolicyRef: &v1beta2.NetworkPolicyReference{
-			Type:      v1beta2.K8sNetworkPolicy,
-			Namespace: "ns1",
-			Name:      "np1",
-			UID:       "id1",
+	tests := []struct {
+		name            string
+		rules           []*types.PolicyRule
+		expectedFlowsFn func(c *client) []binding.Flow
+	}{
+		{
+			name: "multiple K8s NetworkPolicy rules",
+			rules: []*types.PolicyRule{
+				{
+					Direction: v1beta2.DirectionOut,
+					From:      parseAddresses([]string{"192.168.1.40", "192.168.1.50"}),
+					To:        parseAddresses([]string{"0.0.0.0/0"}),
+					FlowID:    uint32(10),
+					TableID:   EgressRuleTable,
+					PolicyRef: &v1beta2.NetworkPolicyReference{
+						Type:      v1beta2.K8sNetworkPolicy,
+						Namespace: "ns1",
+						Name:      "np1",
+						UID:       "id1",
+					},
+				},
+				{
+					Direction: v1beta2.DirectionOut,
+					// conjunctive match flow for nw_src=192.168.1.40 should commit to conjunction 10 and 11.
+					// conjunctive match flow for nw_src=192.168.1.45 should commit to conjunction 11 only.
```

Comment:
what's `192.168.1.45`? Did you mean `192.168.1.51` maybe?

I don't think "commit" is the right verb here, but I'm no specialist. I would suggest "tie" instead.

Reply:
fixed, thanks